Objects/bytearrayobject.c

```diff
@@ -127,40 +127,35 @@ PyObject *
 PyByteArray_FromStringAndSize(const char *bytes, Py_ssize_t size)
 {
     PyByteArrayObject *new;
-    Py_ssize_t alloc;
 
     if (size < 0) {
         PyErr_SetString(PyExc_SystemError,
             "Negative size passed to PyByteArray_FromStringAndSize");
         return NULL;
     }
 
-    /* Prevent buffer overflow when setting alloc to size+1. */
+    /* Prevent buffer overflow when setting alloc to size. */
     if (size == PY_SSIZE_T_MAX) {
```

Comment:
fixme: Change to use new + resize which will do all the needed allocation + resizing + bounds checking (resize is used in __init__ and works on unininitialized bytearray objects). Rely on it to raise the OverflowError.

Reply:
doesn't actually simplify code / diff is equal.